datastream-kafka-connector/src/main/java/com/linkedin/datastream/connectors/kafka/mirrormaker/KafkaMirrorMakerConnector.java

```diff
@@ -313,12 +313,23 @@ public void run() {
       while (!isInterrupted() && !_shutdown) {
         try {
           List<String> newPartitionInfo = getPartitionsInfo(consumer);
-          LOG.debug("Fetch partition info for {}, oldPartitionInfo: {}, new Partition info: {}"
-              , datastream.getName(), _subscribedPartitions, newPartitionInfo);
+          LOG.debug("Fetch Partitions Info for {}, old Partitions Info: {}, new Partitions Info: {}",
+              datastream.getName(), _subscribedPartitions, newPartitionInfo);
 
           if (!ListUtils.isEqualList(newPartitionInfo, _subscribedPartitions)) {
-            LOG.info("Get updated partition info for {}, oldPartitionInfo: {}, new Partition info: {}"
-                , datastream.getName(), _subscribedPartitions, newPartitionInfo);
+            LOG.info("Get updated Partitions Info for {}, old Partitions Info: {}, new Partitions Info: {}",
+                datastream.getName(), _subscribedPartitions, newPartitionInfo);
+
+            List<String> addedTopicPartitions = newPartitionInfo
+                .stream()
+                .filter(topicPartition -> !_subscribedPartitions.contains(topicPartition))
+                .collect(Collectors.toList());
+            List<String> removedTopicPartitions = _subscribedPartitions
+                .stream()
+                .filter(topicPartition -> !newPartitionInfo.contains(topicPartition))
+                .collect(Collectors.toList());
+            LOG.info("TopicPartitions for {} that are to be added: {} and that are to be removed: {}",
```

Comment:
IMO, Splitting this in two lines will make the search easier while debugging.

Reply:
done